pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-headers.adoc

```diff
@@ -0,0 +1,125 @@
+    let headers_state = request_state.into_headers_state().await;
+    let headers_handler = headers_state.handler();
+    let old_value = headers_handler.header("request-header").unwrap_or_default();
+
+    let new_value = "--replaced--";
+    logger::info!("Old request header value: {old_value}, New value: {new_value}");
+    headers_handler.set_header("request-header", new_value);
+}
+
+async fn response_filter(response_state: ResponseState, _config: &Config) {
+    let headers_state = response_state.into_headers_state().await;
+    let headers_handler = headers_state.handler();
+    let old_headers = headers_handler.header("request-header").unwrap_or_default();
+
+    let new_value = vec![("response-header1", "--replaced--"), ("response-header2", "--replaced--")];
+    logger::info!("Old request header value: {old_headers:?}, New value: {new_value:?}");
+    headers_handler.set_headers(new_value);
+}
+----
+
+Envoy handles the method, scheme, path, authority, and status codes as headers. Modify the headers by accessing the request's `:method`, `:scheme`, `:path`, `:authority`, and `:status`. PDK provides methods for ease of access:
+
+[source,Rust]
+----
+async fn request_filter(request_state: RequestState, _config: &Config) {
+    let headers_state = request_state.into_headers_state().await;
+    let method = headers_state.method();
+    let scheme = headers_state.scheme();
+    let authority = headers_state.authority();
+    let path = headers_state.path();
+    ...
+}
+
+async fn response_filter(response_state: ResponseState, _config: &Config) {
+    let headers_state = response_state.into_headers_state().await;
+    let status = headers_state.status_code();
+}
+----
+
+[[read-and-write-request-bodies]]
+== Read and Write Request Bodies
+
+To access the body in both the request and the response, transform the `RequestState` or `ResponseState` to a body state by calling the method `into_body_state()` and awaiting it. If the original state was already transformed into a header state, transform the state into a body state by calling the same function, for example:
+
+[source,Rust]
+----
+let headers_state = request_state.into_headers_state().await;
+let body_state = headers_state.into_body_state().await;
+----
+
+After calling `into_body_state()`, access and manipulate the headers by calling the functions of the `BodyHandler` trait.
+
+[source,Rust]
+----
+pub trait BodyHandler {
+    fn body(&self) -> Vec<u8>;
+}
+----
+
+Because Envoy uses the same buffer to share data from the headers and the body, the policy cannot access the headers and the body at the same time. If the policy must read both:
+
+. Read the headers and save the necessary values in a variable.
+. Read the body.
+
+You can read the headers and then the body in both the response and request.
+
+You cannot modify headers after reading the body. Complete all header modification before reading the body.
+
+For example:
```

Comment:
I'm a little confused. Is this an example of both of the above sentences? Or just the sentence immediately above?

If it's not an example of both, I think it should be:

"You cannot modify headers after reading the body. Complete all header modification before reading the body, for example:"

This pins the example only to that sentence.